src/System.Management.Automation/engine/parser/CharTraits.cs

```diff
@@ -376,11 +376,16 @@ internal static bool ForceStartNewToken(this char c)
 
         // Return true if the character ends the current number token.  This allows the tokenizer
         // to scan '7z' as a single token, but '7+' as 2 tokens.
-        internal static bool ForceStartNewTokenAfterNumber(this char c)
+        internal static bool ForceStartNewTokenAfterNumber(this char c, bool forceEndNumberOnTernaryOperatorChars)
         {
```

Comment:
Instead of a parameter, maybe you should have 2 distinct functions.

Reply:
We need to pass in the parameter even with 2 distinct functions in `CharTraits`, because that bool flag comes from the tokenizer.